rust/src/go/nmstate/nmstate_test.go

```diff
@@ -0,0 +1,20 @@
+package nmstate
+
+import "testing"
+import "github.com/stretchr/testify/assert"
+
+func TestRetrieveNetState(t *testing.T) {
+	nms := nmstate.New()
+	netState, err := nms.RetrieveNetState()
+	assert.NoError(t, err, "must succeed calling retrieve_net_state c binding")
+	assert.NotEmpty(t, netState, "net state should not be empty")
+}
+
+func TestRetrieveNetStateKernelOnly(t *testing.T) {
+	nms := nmstate.New(
+		nms.WithKernelOnly(),
+	)
+	netState, err := nms.RetrieveNetState()
+	assert.NoError(t, err, "must succeed calling retrieve_net_state c binding")
+	assert.NotEmpty(t, netState, "net state should not be empty")
+}
```

Comment:
Don't we want to test apply ?

Reply:
Yes, but we are going to integrate this with the CI later. Let's merge a simple support and basic coverage. Thanks!